### PR TITLE
The stressmodel could use some optimizations to bring dev time down.

### PR DESCRIPTION
**Pull request by brentlintner**
Depends on: https://github.com/zeitspace/GoBikeRouter/pull/39

In terms of local dev, the turn around time has gotten quite high
to where it's maybe `40m` using the documented docker compose db
creation in GoBikeRouter which only supports x86_64, and maybe `20m` if
using native pgrouting. That is if it didn't randomly hang, which was
another issue irrespective (mostly...) of x86_64 emulation.

Same with updating the production and staging databases, which
can take even longer due to some other optimizations that could
be done.

These changes bring the create+seed time to under `4m` with
native osm tooling and pgrouting, and around `10m` when updating
across a network (local -> staging RDS), which should be even
faster when the db is updated via CI/CD.

Extra Note:

I also ended up refactoring/changing a lot in my testing.
The biggest wins are from transactions, segmenting the
rationale, and also fixing a missing `await` when truncating
the rationale table.

**For Review:**

Mainly just checkout [db.js](https://github.com/zeitspace/stressmodel/pull/27/files#diff-d7c9c4e37c37bdbf3fd7007e8e5a111ba1cbfecdd20b76f549a385ebc890e552) for most changes.

## Review Changes

- Catalin's comments
- Catalin: the verbose logging of each batch is pretty verbose (maybe use a one line progress bar pkg, or remove the batch logging for now) Added! Thnkx for the code! 😄 
- Catalin: Revert dbconn back to vars? (less disruptive, and it was just a convenience for gobikerouter db scripts, and I forgot about stressmodel's `.env.example` lol)

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    if (log) await logWayLevelsCount(client, waysTable, stressLevelColumn);
+  } catch (batchError) {
+    console.log('An error happened! Rolling back last transaction... ');
+
+    try {
+      await client.query('ROLLBACK')
+    } catch (rollbackError) {
+      console.error(rollbackError);
+    } finally {
+      throw batchError
```
Comment (catalinzpop) [inline db.js:101]: ;

File: db.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+         4 = ${fourRes.rows[0].count}`
+  );
+}
+
+export async function updateDB(data, options) {
+  const log = !!options.verbose;
+  const rationaleTable = options.rationaleTableName;
+  const updateRationale = !!rationaleTable;
+  const waysTable = options.waysTableName;
+  const stressLevelColumn = options.stressLevelColumnName
```
Comment (catalinzpop) [inline db.js:33]: ;

File: db.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  try {
+    for (let i = 0; i < stressLevels.length; i++) {
+      const stressLevel = stressLevels[i];
+      const ways = data[stressLevel];
+      const batches = chunk(ways, 1000);
+
+      for (let j = 0; j < batches.length; j++) {
+        const batchOfWays = batches[j];
+        const wayIds = map(batchOfWays, (way) => way.id)
```
Comment (catalinzpop) [inline db.js:55]: ;

**Comment by brentlintner** [comment]
[When deployed to staging or production, the database-updater updates the database](https://trello.com/c/2CJN6RtO/548-when-deployed-to-staging-or-production-the-database-updater-updates-the-database)

**Review by peterspenler — APPROVED** [review]
LGTM

**Pull request merged**